packages/next/src/client/components/react-dev-overlay/_experimental/internal/components/ErrorOverlay/ErrorOverlay.tsx

```diff
@@ -0,0 +1,58 @@
+import type { VersionInfo } from '../../../../../../../server/dev/parse-version-info'
+import { Dialog, DialogHeader, DialogBody, DialogContent } from '../Dialog'
+import { Overlay } from '../Overlay'
+import { VersionStalenessInfo } from '../VersionStalenessInfo'
+
+type ErrorOverlayProps = {
+  errorType:
+    | 'Build Error'
+    | 'Runtime Error'
+    | 'Console Error'
+    | 'Unhandled Runtime Error'
+    | 'Missing Required HTML Tag'
+  errorMessage: string | React.ReactNode
+  onClose: () => void
+  versionInfo?: VersionInfo
+  children?: React.ReactNode
+}
+
+export function ErrorOverlay({
```

Comment:
Can give a more presentational flavour to this naming here - call it ErrorOverlayLayout